spec/models/category_spec.rb

```diff
@@ -3,7 +3,16 @@
 RSpec.describe Category, type: :model do
   describe 'validations' do
     it { should validate_presence_of(:name) }
-    it { should validate_uniqueness_of(:name) }
+    it { should validate_uniqueness_of(:name).case_insensitive }
+
+    it 'should not validate name uniqueness with deleted category' do
```

Comment:
Consider refactoring to make it modular category deleted(soft) and a new category
e.g:
```
context 'soft deleted' do
  let(!cat1) { create(:category, name: 'foo') }
  it 'should count deleted category as soft deleted' do
    cat1.destroy
    expect(Category.only_deleted.count).to eq(1)
  end
end
```
Please see http://www.betterspecs.org/#let for more idea